bevm/bevmclient/utils.go

```diff
@@ -0,0 +1,210 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"go.dedis.ch/cothority/v3/bevm"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/onet/v3/log"
+	"golang.org/x/xerrors"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	cli "github.com/urfave/cli"
+)
+
+type userEvmAccount struct {
+	PrivateKey string
+	Nonce      uint64
+}
+
+func writeFile(file string, data []byte, check bool) error {
+	if check {
+		fileInfo, err := os.Stat(file)
+		if !os.IsNotExist(err) {
```

Comment:
I agree it is confusing :), but this would actually not work. `os.IsExist()` and `os.IsNotExist()` are not always the opposite of one another. Here, if the file exists, `os.Stat()` will not return an error, because that is what it expects, so `os.IsExist()` would be `false`. See e.g. [here](https://loganstucki.me/blog/2019-01-17GolangErrorCheckers/) or [here](https://gist.github.com/mastef/05f46d3ab2f5ed6a6787).

There are usually 3 cases to check: the file exists, the file does not exist, some other error happened (such as permissions). Here, to avoid complication, I lumped the "some other error" case together with "the file exists", assuming that if some other error happened it would fail in the rename as well. I can change it to check explicitly for the "other error" case and just fail.

Reply:
Rewrote a bit according to [this](https://goruncode.com/how-to-check-if-a-file-exists/).